RocketChat/Rocket.Chat
# fix: Invalid JSON in Custom Translations setting crashes the UI

## Proposed changes (including videos or screenshots)


- Check if JSON code settings (with `type: code` and `code: application/json`) are valid before saving, or throw an error otherwise;
- Dismiss custom settings when the custom translations JSON provided is invalid (instead of throwing an error and crashing the UI);
- Fix the Custom Translations setting desciption so that the example is shown as a code snippet.



## Issue(s)

Fixes #27846

## Steps to test or reproduce

 - Go to **Workspace > Settings > General > Translations > Custom Translations** and try adding an invalid JSON to the field. An error should be thrown displaying "Invalid JSON provided".
 - Add a valid JSON to the field and save the changes. Then, try erasing the JSON inserted in the field -- this action should be performed succesfully.

## Further comments

[TC-554]

[TC-554]: https://rocketchat.atlassian.net/browse/TC-554?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
